File: engine/src/main/java/client/database/data/GiftData.java
package client.database.data;

public class GiftData {
   private int sn;

   private int ringId;

   private String message;

   private String from;

   public GiftData(int sn, int ringId, String message, String from) {
      this.sn = sn;
      this.ringId = ringId;
      this.message = message;
      this.from = from;
   }

   public int getSn() {
      return sn;
   }

   public int getRingId() {
      return ringId;
   }

   public String getMessage() {
      return message;
   }

   public String getFrom() {
      return from;
   }
}

File: cos/cos-model/src/main/scala/client/database/data/GiftData.scala
package client.database.data

class GiftData(private var _sn: Int, private var _ringId: Int, private var _message: String,
               private var _from: String) {
  def sn: Int = _sn

  def ringId: Int = _ringId

  def message: String = _message

  def from: String = _from
}
